src/rules-engine/ThingConnection.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+const e2p = require('event-to-promise');
+const Settings = require('../models/settings');
+const WebSocket = require('ws');
+
+class ThingConnection {
+  constructor(href, messageHandler) {
+    this.href = href;
+    this.messageHandler = messageHandler;
+    this.onMessage = this.onMessage.bind(this);
+    this.ws = null;
+  }
+
+  async start() {
+    const jwt = await Settings.get('RulesEngine.jwt');
+    const gateway = await Settings.get('RulesEngine.gateway');
+    const wsHref = gateway.replace(/^http/, 'ws') + this.href + '?jwt=' + jwt;
```

Comment:
This is probably OK for now, but note that our current spec defines a links member of the Thing Description which includes the WebSocket endpoint. We may not always be able to assume that the WebSocket URL is the same as the Thing URL. But I realise we don't currently implement the links member and it may be subject to change in the Interest Group.

Reply:
I can definitely change this once we implement the links member